spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/session/SessionAutoConfigurationJdbcTests.java

```diff
@@ -86,4 +88,15 @@ public void customTableName() {
 				.queryForList("select * from FOO_BAR")).isEmpty();
 	}
 
+	@Test
+	public void customTableNameWithDefaultSchemaFails() {
+		this.thrown.expect(BeanCreationException.class);
+		this.thrown.expectMessage(containsString(
+				"Database initializer cannot be configured"));
+		load(Arrays.asList(EmbeddedDataSourceConfiguration.class,
+				DataSourceTransactionManagerAutoConfiguration.class),
+				"spring.session.store-type=jdbc",
+				"spring.session.jdbc.table-name=FOO_BAR");
```

Comment:
Sorry for the confusion but from your previous comments it wasn't obvious as you didn't provide the logical correlation between your _You provided a custom table name? Then you're on your own to create the schema_ and _You provided the path to a script? We'll of course use that to initialize the database_ statements.

I'll update the PR to place the validation logic into the `SessionProperties`. Do you agree that if we don't fail fast, there should be some kind of warning log indicating that initializer was disabled due to particular combination of configuration?